File: 7th Spring Framework & Spring Boot/20211229/boards/src/main/java/com/dto/BbsDto.java
package com.dto;

import lombok.Getter;
import lombok.ToString;

import java.io.Serializable;

@Getter
@ToString
public class BbsDto implements Serializable {

    private int seq;        // 글의 고유번호
    private String id;

    private int ref;        // 그룹번호
    private int step;       // 행번호
    private int depth;      // 깊이

    private String title;
    private String content;
    private String wdate;

    private int del;
    private int readcoount;

    public BbsDto() {}

    /*
        ref : 그룹 번호
        step: 행 번호
        depth: 깊이 번호
        ref - step - depth 순으로
        -------------------------
        세번째 글                                (2 - 0 - 0)
        두번째 글                                (1 - 0 - 0)
          ㄴ 두번째 글의 두번째 답글                 (1 - 1 - 1)
             ㄴ 두번째 글의 두번째 답글의 첫번째 답글   (1 - 2 - 2)
          ㄴ 두번째 글의 첫번째 답글                 (1 - 3 - 1)
        첫번째 글                                (0 - 0 - 0)
    */

    public BbsDto(int seq, String id, int ref, int step, int depth, String title, String content, String wdate, int del, int readcoount) {
        super();
        this.seq = seq;
        this.id = id;
        this.ref = ref;
        this.step = step;
        this.depth = depth;
        this.title = title;
        this.content = content;
        this.wdate = wdate;
        this.del = del;
        this.readcoount = readcoount;
    }

    public BbsDto(String id, String title, String content) {
        this.id = id;
        this.title = title;
        this.content = content;
    }

    public void setSeq(int seq) {
        this.seq = seq;
    }

    public void setId(String id) {
        this.id = id;
    }

    public void setContent(String content) {
        this.content = content;
    }
}

File: 12th Android+SpringBoot/BBSSystem/app/src/main/java/com/hwangduil/bbssystem/dto/BbsDto.kt
package com.hwangduil.bbssystem.dto

import android.os.Parcel
import android.os.Parcelable

class BbsDto(var seq:Int,
             var id:String?,
             var ref:Int,
             var step:Int,
             var depth:Int,
             var title:String?,
             var content:String?,
             var wdate:String?,
             var del:Int,
             var readCount:Int) : Parcelable {

    constructor(parcel: Parcel) : this(parcel.readInt(),
                                       parcel.readString(),
                                       parcel.readInt(),
                                       parcel.readInt(),
                                       parcel.readInt(),
                                       parcel.readString(),
                                       parcel.readString(),
                                       parcel.readString(),
                                       parcel.readInt(),
                                       parcel.readInt())

    override fun writeToParcel(parcel: Parcel, p1: Int) {
        parcel.writeInt(seq)
        parcel.writeString(id)
        parcel.writeInt(ref)
        parcel.writeInt(step)
        parcel.writeInt(depth)
        parcel.writeString(title)
        parcel.writeString(content)
        parcel.writeString(wdate)
        parcel.writeInt(del)
        parcel.writeInt(readCount)
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun toString(): String {
        return "BbsDto(seq=$seq, id=$id, ref=$ref, step=$step, depth=$depth, title=$title, content=$content, wdate=$wdate, del=$del, readCount=$readCount)"
    }

    companion object CREATOR : Parcelable.Creator<BbsDto> {
        override fun createFromParcel(parcel: Parcel): BbsDto {
            return BbsDto(parcel)
        }

        override fun newArray(size: Int): Array<BbsDto?> {
            return arrayOfNulls(size)
        }
    }


}